src/ewatercycle/plugins/hype/model.py

```diff
@@ -190,6 +165,20 @@ def get_value_as_xarray(self, name: str) -> xr.DataArray:
         """
         raise NotImplementedError("Hype coordinates cannot be mapped to grid")
 
+    def get_latlon_grid(self, name: str) -> tuple[Any, Any, Any]:
+        raise NotImplementedError("Hype coordinates cannot be mapped to grid")
+
+    def get_parameters(self) -> Iterable[Tuple[str, Any]]:
```

Comment:
@BSchilperoort shy did you change the parameters from a property called `parameters` to a method called `get_parameters`? Is this to do with what you said about pydantic?

Note that we want to use parameter akin to pymt: https://pymt.readthedocs.io/en/latest/usage.html#model-setup

Reply:
I now see that `*Model.parameters` is defined as a variable property in `BaseModel`. I think it conflicted with defining the parameters as `@property` in the specific model implementations. 

I guess it's best to change how the BaseModel implements that.


Side note: I've found it a bit confusing that our `BaseModel` has the same class name as the pydantic `BaseModel` :confused: 
```py
class BaseModel(pydantic.BaseModel, abc.ABC):
    ...
```

Reply:
Renamed to eWaterCycleModel in https://github.com/eWaterCycle/ewatercycle/pull/361/commits/998b22684ae1a3e3f624ba97bd3fd282712d6ee9